src/semantic_world/geometry.py

```diff
@@ -476,33 +456,88 @@ def get_points(self) -> List[Point3]:
                 for z in (self.min_z, self.max_z)]
 
     @classmethod
-    def from_min_max(cls, min_point: Point3, max_point: Point3) -> Self:
+    def from_min_max(
+        cls, reference_frame: Body, min_point: Point3, max_point: Point3
+    ) -> Self:
```

Comment:
Point3 has a reference frame. so you should probably not have it as a parameter in from_min_max, but verify that the two points have the same reference frame and the use it.